pkg/primaza/controlplane/application_namespaces.go

```diff
@@ -127,3 +128,36 @@ func PushServiceCatalogToApplicationNamespaces(ctx context.Context, sc primazaio
 
 	return nil
 }
+
+func DeleteServiceBindingAndSecretFromNamespaces(ctx context.Context, cli client.Client, sc primazaiov1alpha1.ServiceClaim, namespaces []string) error {
+	var errs []error
+	for _, ns := range namespaces {
+		sb := &primazaiov1alpha1.ServiceBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      sc.Name,
+				Namespace: ns,
+			},
+		}
+
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      sc.Name,
+				Namespace: ns,
+			},
+		}
+
+		if err := cli.Delete(ctx, sb, &client.DeleteOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				errs = append(errs, err)
+			}
+
+		}
+		if err := cli.Delete(ctx, secret, &client.DeleteOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				errs = append(errs, err)
+			}
+		}
```

Comment:
another approach may be to leverage on Ownership for this. It would reduce the probability of having an error and having to reconcile again. We should modify the `controlplane.pushServiceBindingToNamespace` to push the ServiceBinding before creating the Secret and enriching the Secret with Ownership details from ServiceBinding.

PTAL: https://github.com/filariow/kid/blob/4e10389547c1bd78e0526cb80a8d2dd8c5671681/pkg/kube/secret.go#L72

After applied this change we should just delete the ServiceBinding. This would also remove the requirement for the `delete` permission added in `kubernetesapi.py`.

WDYT?

Reply:
Yeah i believe, that is better as we won't need to explicitly set the permission to delete secret. I have made the change in pushservicebinding method to create secret with owner reference set to service binding uid.